### PR TITLE
Make docker image tag dynamic

### PR DESCRIPTION
**Pull request by utkarsharma2**
# Description
## What is the current behavior?
Every new docker image is labeled as latest and if we have two parallel branches that run benchmark one might conflict with another branch. We should have new labels per commit basis.

closes: #887

## What is the new behavior?
Modified Makefile to use git sha instead of the latest tag.

## Does this introduce a breaking change?
Nope

### Checklist
- [ ] Created tests which fail without the change (if possible)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

**Review by pankajkoti — COMMENTED** [review]
This would mean every time we make local changes and want to run benchmark, we would have to commit right? How about instead generating a hash so that although 2 developers have the same git commit hash on local (before they commit), so that they have different tags?

**Review by tatiana — APPROVED** [review]

**Review by pankajastro — APPROVED** [review]

**Comment by pankajastro** [comment]
> This would mean every time we make local changes and want to run benchmark, we would have to commit right? How about instead generating a hash so that although 2 developers have the same git commit hash on local (before they commit), so that they have different tags?

Benchmark we don't run so frequently so I'm ok with this change also otherwise maybe we can take the current time in milliseconds + gittag WDYT?

File: python-sdk/tests/benchmark/infrastructure/kubernetes/job.yaml [hunk 1]
```diff
@@ -10,7 +10,7 @@ spec:
     spec:
       containers:
       - name: benchmark
-        image: gcr.io/astronomer-dag-authoring/benchmark
+        image: gcr.io/astronomer-dag-authoring/benchmark:%VERSION%
```
Comment (feluelle) [inline python-sdk/tests/benchmark/infrastructure/kubernetes/job.yaml:13]: can you not access the env var here?

File: python-sdk/tests/benchmark/Makefile [hunk 2]
```diff
@@ -59,7 +59,8 @@ run_job:
 	@kubectl apply -f infrastructure/kubernetes/namespace.yaml
 	@kubectl apply -f infrastructure/kubernetes/postgres.yaml
 	@kubectl apply -f infrastructure/kubernetes/postgres_svc.yaml
-	@kubectl create -f infrastructure/kubernetes/job.yaml
+	@sed s/%VERSION%/${GIT_HASH}/g infrastructure/kubernetes/job.yaml > infrastructure/kubernetes/job_final.yaml
+	@kubectl create -f infrastructure/kubernetes/job_final.yaml
```
Comment (feluelle) [inline python-sdk/tests/benchmark/Makefile:63]: This looks like a hack to me. There has to be another way of passing values from outside into the yaml file.

File: python-sdk/tests/benchmark/infrastructure/kubernetes/job.yaml (hunk 1 above)
Comment (utkarsharma2) [reply python-sdk/tests/benchmark/infrastructure/kubernetes/job.yaml:13]: @feluelle As per my understanding. We can't. But do let me know if you think there is a way

File: python-sdk/tests/benchmark/Makefile (hunk 2 above)
Comment (utkarsharma2) [reply python-sdk/tests/benchmark/Makefile:63]: @feluelle This is not the ideal solution but is there an easier alternative? 

I considered `yq` to replace the yaml file but then again you have to create a new file and you need to install `yq`.

Comment (kaxil) [reply python-sdk/tests/benchmark/Makefile:63]: There is another hack - https://stackoverflow.com/a/56240712 :)

Comment (kaxil) [reply python-sdk/tests/benchmark/Makefile:63]: But it is just a hack -- you can't template k8s yaml without Helm natively I think

**Review by kaxil — APPROVED** [review]

File: python-sdk/tests/benchmark/Makefile (hunk 2 above)
Comment (feluelle) [reply python-sdk/tests/benchmark/Makefile:63]: Right - you would have to use helm.

**Pull request merged**